spec/models/user_spec.rb

```diff
@@ -190,28 +209,44 @@
   end
 
   context "when updating name" do
-    it "is valid to leave name unset" do
-      user = create :user, :with_no_name
-      user.name = nil
-      expect(user.valid?).to be true
+    let(:user) { create :user, :with_no_name, role: :trial }
+
+    context "when user has been created with a trial account" do
+      it "is valid to leave name unset" do
+        user.name = nil
+        expect(user).to be_valid
+      end
+    end
+
+    described_class.roles.each_key do |role|
+      context "when user somehow has no name" do
+        let(:user) do
+          user = build(:user, :with_no_name, role:)
+          user.save!(validate: false)
+          user
+        end
+
+        it "is valid to leave name unset" do
+          user.name = nil
+          expect(user).to be_valid
+        end
+      end
     end
 
     it "is not valid to unset name if it is already set" do
-      user = create :user, name: "Test User"
+      user.update!(name: "Test User")
       user.name = nil
-      expect(user.valid?).to be false
+      expect(user).not_to be_valid
     end
 
     it "is not valid to leave name unset if changing role to editor" do
-      user = create :user, :with_no_name, role: :trial
       user.role = :editor
-      expect(user.valid?).to be false
+      expect(user).not_to be_valid
     end
 
     it "is not valid to leave name unset if changing role to super admin" do
-      user = create :user, :with_no_name, role: :trial
       user.role = :super_admin
-      expect(user.valid?).to be false
+      expect(user).not_to be_valid
```

Comment:
Only a suggestion, any thoughts on using this instead? I find it easier to read but its a personal preference, either way is totally fine.

```suggestion
      expect(user).to be_invalid
```

Reply:
Works for me, I'll do that 👍